cloud/scope/powervs_cluster_test.go

```diff
@@ -153,3 +158,1457 @@ func TestNewPowerVSClusterScope(t *testing.T) {
+			return mockResourceController, nil
+		}
+		clusterScope, err := NewPowerVSClusterScope(clusterScopeParams)
+		g.Expect(err).To(BeNil())
+
+		instance, requeue, err := clusterScope.isServiceInstanceExists()
+		g.Expect(instance).To(Equal(""))
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+	t.Run("When isServiceInstanceExists returns success", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.ResourceControllerFactory = func() (resourcecontroller.ResourceController, error) {
+			mockResourceController.EXPECT().GetServiceInstance(gomock.Any(), gomock.Any(), gomock.Any()).Return(&resourcecontrollerv2.ResourceInstance{GUID: ptr.To("guid"), Name: ptr.To("instance"), State: ptr.To("active")}, nil)
+			return mockResourceController, nil
+		}
+		clusterScope, err := NewPowerVSClusterScope(clusterScopeParams)
+		g.Expect(err).To(BeNil())
+
+		instance, requeue, err := clusterScope.isServiceInstanceExists()
+		g.Expect(instance).To(Equal("guid"))
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).To(BeNil())
+	})
+}
+
+func TestCreateServiceInstance(t *testing.T) {
+	var (
+		mockResourceController *mockRC.MockResourceController
+		mockCtrl               *gomock.Controller
+	)
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockResourceController = mockRC.NewMockResourceController(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When resource group is not set", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.ResourceGroup = nil
+		clusterScope, err := NewPowerVSClusterScope(clusterScopeParams)
+		g.Expect(err).To(BeNil())
+
+		instance, err := clusterScope.createServiceInstance()
+		g.Expect(instance).To(BeNil())
+		g.Expect(err).NotTo(BeNil())
+	})
+	t.Run("When zone is not set", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScope, err := NewPowerVSClusterScope(clusterScopeParams)
+		g.Expect(err).To(BeNil())
+		clusterScope.IBMPowerVSCluster.Spec.Zone = nil
+
+		instance, err := clusterScope.createServiceInstance()
+		g.Expect(instance).To(BeNil())
+		g.Expect(err).NotTo(BeNil())
+	})
+	t.Run("When CreateResourceInstance returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.ResourceControllerFactory = func() (resourcecontroller.ResourceController, error) {
+			mockResourceController.EXPECT().CreateResourceInstance(gomock.Any()).Return(nil, nil, fmt.Errorf("intentional error"))
```

Comment:
might be good to have a better error message for readability and to keep in sync with previously added UTs